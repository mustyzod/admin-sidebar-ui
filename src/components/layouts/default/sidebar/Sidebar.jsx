import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import { SideBarProfile } from './SideBarTop';
import styled from "styled-components";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { MenuLists } from './MenuList';

//  Set default route
let currentRoute = "/";
let currentSubRoute = "/";
let currentRoute2 = "/";

// react style
const isActive = {
    width: "auto",
    // background: "rgba(255, 255, 255, .05)",
    // color: "#8d0000",
    cursor: "pointer",
    webkitTransitionDelay: "0s",
    borderRight: "8px solid #8d0000",

    boxShadow: "rgba(0, 0, 0, .25) 0 1px 0, inset rgba(255, 255, 255, .16) 0 1px 0",
    textShadow: "rgba(0, 0, 0, .25) 0 - 1px 0!important",
    background: "linear - gradient(#1c1c1c, #1c1c1c), #8d0000!important"
}
//  react style ends

//  styled component block
const LeftCol = styled.div`
    min-height: 100%;
    width: 230px;
    padding: 0;
    position: fixed;
    display: -ms-flexbox;
    display: flex;
    z-index: 1;
        @media (max-width: 768px) {
        width:70px;
        overflow-x: visible;
    }
`;
const SidebarContainer = styled.div`
    height:100vh;
    width:230px;
    top:0;
    // position:fixed;
    z-index:2;
    background-color:#252529;
    color:#fff;
    font-family: "Roboto", sans-serif;
    display:flex;
    flex-direction:column;

    overflow-x: hidden;
    @media (max-width: 768px) {
        width:70px;
        overflow-x: visible;
    }
`;

const MenuDropdownContainer = styled.div`
    display:${props => props.display || "none"} ;
    z-index:99;
    @media (max-width: 768px) {
        background: #1c1c1c;
        position:absolute;
        margin-top: 1rem;
        margin-left: 4.8rem;
        
        &:before {
            content: "";
            height: 8px;
            left: -7px;
            margin-top: 5px;
            position: absolute;
            height: 0;
            border-top: 8px solid transparent;
            border-bottom: 8px solid transparent;
            border-left: 8px solid #8d0000;
        }
    }
`;

const SideMenu = styled(Link)`
    color:darkgrey;
    text-decoration:none;

    display: flex;
    flex-direction: column;
    width:auto;
    position:relative;

    &:hover ${MenuDropdownContainer} {
        display: block;
    }
`;
const SideMenuDrop = styled.div`
    ${props => (props.to === '/' + currentRoute) ? isActive : ""}
    color:#fff;
    text-decoration:none;
    display: flex;
    flex-direction: column;
    width:auto;
    position:relative;
    @media (max-width: 769px) {
        &:hover ${MenuDropdownContainer} {
            display: block;
        }
    }
    // make dropdown active on wide screen
    @media (min-width: 769px) {
        ${props => (props.to === '/' + currentRoute) ? MenuDropdownContainer + '{display: block}' : "none"};
    }
`;

const SideMenuItem = styled.div`
    padding: .7rem 0 .7rem .7rem ; 
    width: auto; 
    text-align: left;
    &:hover{
        background: rgba(255,255,255,.05);
        color:#8d0000;
        cursor: pointer;
        -webkit-transition-delay: 0s;
        // border-right: 8px solid #8d0000;
    }
`;

const SideMenuIcon = styled.span`
    font-size: ${props => props.fontSize ? props.fontSize : "1.2rem"}; 
    padding: 1rem;
    @media (max-width: 768px) {
        font-size: 2rem;
        padding-left:.5rem;
    }
`;

const SidebarMenuItemLabel = styled.span`
    font-family: "Helvetica Neue",Roboto,Arial,"Droid Sans",sans-serif;
    font-size:16px;
    font-weight: 900;
    line-height:1.471;
    text-align:left;
    padding: 12px 0px;
    z-index: 2;
    position: relative;
    -webkit-transition: border-radius .1s linear, background .1s linear, z-index 0s linear;
        -webkit-transition-delay: .8s;

    @media (max-width: 768px) {
        display: none;
    }
`;

const MenuDropdownText = styled(Link)`
    display:flex;
    color:darkgrey;
    font-weight:600;
    text-decoration:none;
    padding:.5rem 4rem;
    font-size: 14px;
    list-style:none;
    text-align:left;
    &:hover{
        background: rgba(255,255,255, 0.05);
        color: #8d0000;
        cursor: pointer;
    }

    &:before {
        background: #425668;
        bottom: auto;
        content: "";
        height: 8px;
        left: 18px;
        margin-top: 4px;
        margin-left:25px;
        position: absolute;
        right: auto;
        width: 8px;
        z-index: 1;
        border-radius: 50%;
    }
    &:after{
        border-left: 1px solid #425668;
        bottom: 0;
        content: "";
        left: 21px;
        margin-left:25px;
        margin-top:-10px;
        position: absolute;
        top: 8px;
    }
    @media (max-width: 768px) {
        padding:.5rem 1rem;
        &:before {
            display:none
        }
        &:after{
            display:none
        }
    }
`;

const SubMenuDropdownContainer = styled.div`
    display:${props => ((props.display === 'block') || (props.url === '/' + currentRoute + '/' + currentRoute2)) ? props.display : "none"};
    background:${props => ((props.display === 'block') || (props.url === '/' + currentRoute + '/' + currentRoute2)) ? "#504e4e26" : "none"};
    z-index:99;
    // background:#504e4e26;
    @media (max-width: 768px) {
        background: #1c1c1c;
        position:absolute;
        width:100%;
        margin-top: 0rem;
        margin-left: 5.4rem;
        
        &:before {
            content: "";
            height: 8px;
            left: -7px;
            margin-top: 5px;
            position: absolute;
            height: 0;
            border-top: 8px solid transparent;
            border-bottom: 8px solid transparent;
            border-left: 8px solid #8d0000;
        }
    }
`;

const SubMenuDropdownText = styled(Link)`
    display:flex;
    white-space: nowrap;

    color:darkgrey;
    font-weight:600;
    text-decoration:none;
    padding:.5rem 5.5rem;
    font-size: 12px;
    list-style:none;
    text-align:left;

    

    &:hover{
        background: rgba(255,255,255, 0.05);
        color: #8d0000;
        cursor: pointer;
    }

    &:before {
        background: #425668;
        bottom: auto;
        content: "";
        height: 7px;
        width: 7px;
        left: 28px;
        margin-top: 4px;
        margin-left:40px;
        position: absolute;
        right: auto;
        z-index: 1;
        border-radius: 50%;
    }
        &:after{
        border-left: 1px solid #425668;
        bottom: 0;
        content: "";
        left: 21px;
        margin-left: 50px;
        margin-top: -10px;
        position: absolute;
        top: 40px;
    }
    @media (max-width: 768px) {
        padding:.5rem 1rem;
        &:before {
            display:none
        }
        &:after{
            display:none
        }
    }
`;

const SideSubMenuDrop = styled.div`
    ${props => (props.to === '/' + currentRoute) ? isActive : ""}
    color:#ffffff;
    text-decoration:none;

    background:#1c1c1c;

    display: flex;
    flex-direction: column;
    width:auto;
    position:relative;
    @media (max-width: 769px) {
        &:hover ${SubMenuDropdownContainer} {
            display: block;
        }
    }
    // make dropdown active on wide screen
    @media (min-width: 769px) {
        ${props => (props.to === currentSubRoute) ? SubMenuDropdownContainer + '{display: block}' : "none"};
    }
`;

const MenuChevronSpan = styled.span`
    @media (min-width: 769px) {
        &:before {
            bottom: auto;
            // content: "";
            ${props => (props.show.length > 0) ? 'content: ""' : ''} ; //show only when their is a child
            left: 18px;
            // margin-top: 12px;
            margin-top: 1.2rem;
            margin-left:185px;
            position: absolute;
            right: auto;
            width:0px;
            border:none;
            border-top:7px solid darkgrey;
            // border-bottom:5px solid #fff;
            border-left:6px solid rgba(0,0,0,0);
            border-right:6px solid rgba(0,0,0,0);
            }
    }
`;

const SubMenuChevronSpan = styled.span`
    @media (min-width: 769px) {
        &:before {
            bottom: auto;
            // content: "";
            ${props => (props.show.length > 0) ? 'content: ""' : ''} ; //show only when their is a child
            left: 18px;
            margin-top: 12px;
            margin-left:185px;
            position: absolute;
            right: auto;
            width:0px;
            border:none;
            border-top:5px solid darkgrey;
            // border-bottom:5px solid #fff;
            border-left:5px solid rgba(0,0,0,0);
            border-right:5px solid rgba(0,0,0,0);
        }
    }
`;
//  styled component block ends

const Sidebar = () => {
    let location = useLocation();
    currentRoute = location.pathname.split('/')[1]; //  Get page location and use it to set the active link
    currentSubRoute = location.pathname;
    currentRoute2 = location.pathname.split('/')[2];
    return (
        <LeftCol>
            <SidebarContainer>
                <SideBarProfile />
                {
                    MenuLists.map(menulist => {
                        return (
                            <React.Fragment key={menulist.id}>
                                <SideMenuDrop to={menulist.url}>
                                    <SideMenu to={menulist.url}>
                                        <MenuChevronSpan show={menulist.content}>
                                            <SideMenuItem>
                                                <SideMenuIcon>
                                                    <FontAwesomeIcon icon={menulist.icon} />
                                                </SideMenuIcon>
                                                <SidebarMenuItemLabel>{menulist.label}</SidebarMenuItemLabel>
                                            </SideMenuItem>
                                        </MenuChevronSpan>
                                    </SideMenu>

                                    <MenuDropdownContainer>
                                        {
                                            menulist.content.map(firstLevel => {
                                                return (
                                                    <SideSubMenuDrop key={firstLevel.id} to={firstLevel.url}>
                                                        <SubMenuChevronSpan show={firstLevel.content} url={firstLevel.url}>
                                                            <MenuDropdownText to={firstLevel.url}>{firstLevel.label}</MenuDropdownText>
                                                        </SubMenuChevronSpan>
                                                        <SubMenuDropdownContainer url={firstLevel.url} >
                                                            {
                                                                firstLevel.content.map(secondLevel => {
                                                                    return (
                                                                        <React.Fragment key={secondLevel.id}>
                                                                            <SubMenuDropdownText to={secondLevel.url}>{secondLevel.label}</SubMenuDropdownText>
                                                                        </React.Fragment>
                                                                    )
                                                                })
                                                            }
                                                        </SubMenuDropdownContainer>
                                                    </SideSubMenuDrop>
                                                )
                                            })
                                        }
                                    </MenuDropdownContainer>
                                </SideMenuDrop>
                            </React.Fragment>
                        )
                    })
                }
            </SidebarContainer >
        </LeftCol>
    )
}
export default Sidebar
