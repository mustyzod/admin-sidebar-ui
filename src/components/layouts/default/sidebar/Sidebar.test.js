import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Sidebar from './Sidebar';


describe('My Test Suite', () => {
    it('My Test Case', () => {
        expect(true).toEqual(true);
    });
});

describe('Make a screenshot for sidebar', () => {
    test('snapshot renders', () => {
        const component = renderer.create(
            <Router>
                <Sidebar />
            </Router>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});