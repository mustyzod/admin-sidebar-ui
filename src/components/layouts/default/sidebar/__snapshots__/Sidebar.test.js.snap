// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make a screenshot for sidebar snapshot renders 1`] = `
<div
  className="sc-AxhCb kFKcQM"
>
  <div
    className="sc-AxhUy izvOWi"
  >
    <div
      className="sc-AxiKw lmzsOo"
    >
      <div
        className="sc-AxjAm gklpOJ"
      >
        <img
          alt="..."
          src="https://graph.facebook.com/1121533047863818/picture?type=large"
          style={
            Object {
              "background": "#8d0000",
              "border": "1px solid rgba(52, 73, 94, .44)",
              "borderRadius": "50%",
              "marginLeft": "15%",
              "marginTop": "20px",
              "padding": "4px",
              "position": "inherit",
              "width": "70%",
              "zIndex": "1000",
            }
          }
        />
      </div>
      <div
        className="sc-AxirZ dpQjXU"
      >
        <p
          style={
            Object {
              "fontSize": "0.8rem",
              "fontWeight": "600",
            }
          }
        >
          Welcome,
          <br />
          Sodruldeen Mustapha
        </p>
        <p
          style={
            Object {
              "color": "#9b9b9b",
              "fontSize": "0.9rem",
              "fontWeight": "900",
            }
          }
        >
          Team Leader - Ä°nformation Technologies
        </p>
        <p
          style={
            Object {
              "color": "#8d0000",
              "fontSize": "1.8rem",
              "fontWeight": "900",
            }
          }
        >
          27345
        </p>
      </div>
    </div>
    <div
      className="sc-Axmtr iCYxnj"
      to="/dashboard"
    >
      <a
        className="sc-AxheI eUVfjG"
        href="/dashboard"
        onClick={[Function]}
      >
        <span
          className="sc-fznZeY itDcay"
        >
          <div
            className="sc-AxmLO zLHqk"
          >
            <span
              className="sc-fzozJi kDtWOF"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-rocket fa-w-16 "
                data-icon="rocket"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505.12019,19.09375c-1.18945-5.53125-6.65819-11-12.207-12.1875C460.716,0,435.507,0,410.40747,0,307.17523,0,245.26909,55.20312,199.05238,128H94.83772c-16.34763.01562-35.55658,11.875-42.88664,26.48438L2.51562,253.29688A28.4,28.4,0,0,0,0,264a24.00867,24.00867,0,0,0,24.00582,24H127.81618l-22.47457,22.46875c-11.36521,11.36133-12.99607,32.25781,0,45.25L156.24582,406.625c11.15623,11.1875,32.15619,13.15625,45.27726,0l22.47457-22.46875V488a24.00867,24.00867,0,0,0,24.00581,24,28.55934,28.55934,0,0,0,10.707-2.51562l98.72834-49.39063c14.62888-7.29687,26.50776-26.5,26.50776-42.85937V312.79688c72.59753-46.3125,128.03493-108.40626,128.03493-211.09376C512.07526,76.5,512.07526,51.29688,505.12019,19.09375ZM384.04033,168A40,40,0,1,1,424.05,128,40.02322,40.02322,0,0,1,384.04033,168Z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
            <span
              className="sc-fzoLsD kwsQVf"
            >
              Dashboard
            </span>
          </div>
        </span>
      </a>
      <div
        className="sc-AxgMl hiBtdB"
      />
    </div>
    <div
      className="sc-Axmtr iCYxnj"
      to="/order"
    >
      <a
        className="sc-AxheI eUVfjG"
        href="/order"
        onClick={[Function]}
      >
        <span
          className="sc-fznZeY foxNFp"
        >
          <div
            className="sc-AxmLO zLHqk"
          >
            <span
              className="sc-fzozJi kDtWOF"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-shopping-basket fa-w-18 "
                data-icon="shopping-basket"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M576 216v16c0 13.255-10.745 24-24 24h-8l-26.113 182.788C514.509 462.435 494.257 480 470.37 480H105.63c-23.887 0-44.139-17.565-47.518-41.212L32 256h-8c-13.255 0-24-10.745-24-24v-16c0-13.255 10.745-24 24-24h67.341l106.78-146.821c10.395-14.292 30.407-17.453 44.701-7.058 14.293 10.395 17.453 30.408 7.058 44.701L170.477 192h235.046L326.12 82.821c-10.395-14.292-7.234-34.306 7.059-44.701 14.291-10.395 34.306-7.235 44.701 7.058L484.659 192H552c13.255 0 24 10.745 24 24zM312 392V280c0-13.255-10.745-24-24-24s-24 10.745-24 24v112c0 13.255 10.745 24 24 24s24-10.745 24-24zm112 0V280c0-13.255-10.745-24-24-24s-24 10.745-24 24v112c0 13.255 10.745 24 24 24s24-10.745 24-24zm-224 0V280c0-13.255-10.745-24-24-24s-24 10.745-24 24v112c0 13.255 10.745 24 24 24s24-10.745 24-24z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
            <span
              className="sc-fzoLsD kwsQVf"
            >
              Order
            </span>
          </div>
        </span>
      </a>
      <div
        className="sc-AxgMl hiBtdB"
      >
        <div
          className="sc-fznKkj kKkxzF"
          to="/order/opened"
        >
          <span
            className="sc-fzokOt cvVYij"
          >
            <a
              className="sc-fzpans faSmpZ"
              href="/order/opened"
              onClick={[Function]}
            >
              Opened
            </a>
          </span>
          <div
            className="sc-fzplWN ezDLlV"
          />
        </div>
        <div
          className="sc-fznKkj kKkxzF"
          to="/order/closed"
        >
          <span
            className="sc-fzokOt gsHlRo"
          >
            <a
              className="sc-fzpans faSmpZ"
              href="/order/closed"
              onClick={[Function]}
            >
              Closed
            </a>
          </span>
          <div
            className="sc-fzplWN ezDLlV"
          >
            <a
              className="sc-fznyAO bcyMbe"
              href="/order/closed/submenu"
              onClick={[Function]}
            >
              Sub Menu
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-Axmtr iCYxnj"
      to="/deliveries"
    >
      <a
        className="sc-AxheI eUVfjG"
        href="/deliveries"
        onClick={[Function]}
      >
        <span
          className="sc-fznZeY itDcay"
        >
          <div
            className="sc-AxmLO zLHqk"
          >
            <span
              className="sc-fzozJi kDtWOF"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-truck-loading fa-w-20 "
                data-icon="truck-loading"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M50.2 375.6c2.3 8.5 11.1 13.6 19.6 11.3l216.4-58c8.5-2.3 13.6-11.1 11.3-19.6l-49.7-185.5c-2.3-8.5-11.1-13.6-19.6-11.3L151 133.3l24.8 92.7-61.8 16.5-24.8-92.7-77.3 20.7C3.4 172.8-1.7 181.6.6 190.1l49.6 185.5zM384 0c-17.7 0-32 14.3-32 32v323.6L5.9 450c-4.3 1.2-6.8 5.6-5.6 9.8l12.6 46.3c1.2 4.3 5.6 6.8 9.8 5.6l393.7-107.4C418.8 464.1 467.6 512 528 512c61.9 0 112-50.1 112-112V0H384zm144 448c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
            <span
              className="sc-fzoLsD kwsQVf"
            >
              Deliveries
            </span>
          </div>
        </span>
      </a>
      <div
        className="sc-AxgMl hiBtdB"
      />
    </div>
    <div
      className="sc-Axmtr iCYxnj"
      to="/extra"
    >
      <a
        className="sc-AxheI eUVfjG"
        href="/extra"
        onClick={[Function]}
      >
        <span
          className="sc-fznZeY foxNFp"
        >
          <div
            className="sc-AxmLO zLHqk"
          >
            <span
              className="sc-fzozJi kDtWOF"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-shopping-basket fa-w-18 "
                data-icon="shopping-basket"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M576 216v16c0 13.255-10.745 24-24 24h-8l-26.113 182.788C514.509 462.435 494.257 480 470.37 480H105.63c-23.887 0-44.139-17.565-47.518-41.212L32 256h-8c-13.255 0-24-10.745-24-24v-16c0-13.255 10.745-24 24-24h67.341l106.78-146.821c10.395-14.292 30.407-17.453 44.701-7.058 14.293 10.395 17.453 30.408 7.058 44.701L170.477 192h235.046L326.12 82.821c-10.395-14.292-7.234-34.306 7.059-44.701 14.291-10.395 34.306-7.235 44.701 7.058L484.659 192H552c13.255 0 24 10.745 24 24zM312 392V280c0-13.255-10.745-24-24-24s-24 10.745-24 24v112c0 13.255 10.745 24 24 24s24-10.745 24-24zm112 0V280c0-13.255-10.745-24-24-24s-24 10.745-24 24v112c0 13.255 10.745 24 24 24s24-10.745 24-24zm-224 0V280c0-13.255-10.745-24-24-24s-24 10.745-24 24v112c0 13.255 10.745 24 24 24s24-10.745 24-24z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
            <span
              className="sc-fzoLsD kwsQVf"
            >
              Extra
            </span>
          </div>
        </span>
      </a>
      <div
        className="sc-AxgMl hiBtdB"
      >
        <div
          className="sc-fznKkj kKkxzF"
          to="/extra/opened"
        >
          <span
            className="sc-fzokOt cvVYij"
          >
            <a
              className="sc-fzpans faSmpZ"
              href="/extra/opened"
              onClick={[Function]}
            >
              Opened
            </a>
          </span>
          <div
            className="sc-fzplWN ezDLlV"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
